specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2021-08-01/apimschema.json

```diff
@@ -0,0 +1,328 @@
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/schemas": {
+      "get": {
+        "tags": [
+          "Schema"
+        ],
+        "operationId": "GlobalSchema_ListByService",
+        "description": "Lists a collection of schemas registered with service instance.",
+        "x-ms-examples": {
+          "ApiManagementListSchemas": {
+            "$ref": "./examples/ApiManagementListGlobalSchemas.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "|     Field     |     Usage     |     Supported operators     |     Supported functions     |</br>|-------------|-------------|-------------|-------------|</br>| name | filter | ge, le, eq, ne, gt, lt | substringof, contains, startswith, endswith |</br>"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Lists a collection of Schema entities.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/GlobalSchemaCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "./definitions.json#/definitions/GlobalSchemaContract"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/schemas/{schemaId}": {
+      "head": {
```

Comment:
>head [](http://example.com/codeflow?start=7&length=4)

ARM won't route HEAD requests to RPs. Etag is a top level property of ARM resources. You need to return it in the GET resource response.

[https://github.com/Azure/azure-resource-manager-rpc/blob/d9c0275c7ada1a65f25ec85d88b3f8fe0bf3dc1e/v1.0/resource-api-reference.md#response-3](https://github.com/Azure/azure-resource-manager-rpc/blob/d9c0275c7ada1a65f25ec85d88b3f8fe0bf3dc1e/v1.0/resource-api-reference.md#response-3)

[https://github.com/Azure/azure-resource-manager-rpc/blob/d9c0275c7ada1a65f25ec85d88b3f8fe0bf3dc1e/v1.0/Addendum.md#etags-for-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/d9c0275c7ada1a65f25ec85d88b3f8fe0bf3dc1e/v1.0/Addendum.md#etags-for-resources)

Reply:
I take it back. ARM will rout HEAD requests to RPs.